src/settings/toml/environment.rs

```diff
@@ -17,3 +18,14 @@ pub struct Environment {
     #[serde(rename = "kv-namespaces")]
     pub kv_namespaces: Option<Vec<KvNamespace>>,
 }
+
+impl Environment {
+    pub fn route_config(&self) -> Result<RouteConfig, failure::Error> {
+        Ok(RouteConfig {
+            workers_dev: self.workers_dev,
+            route: self.route.clone(),
```

Comment:
we do, later. this is kind of an intermediate struct; DeployTarget is what we ultimately put together and send to Publish, and its zoned version does contain JUST a routes vec.

Reply:
why not do it when creating the struct, and have a special deserializer that handles converting `route` -> `routes` when only one route is present? I think this would reduce confusion and inconsistency :)

Reply:
so, the way this logic goes, this is valid:

```toml
route = ""
routes = ["hostname.tld"]
```
i'm not 100% certain how to write the serializer that makes this okay. i'll try in some refactors to make the "RouteConfig" struct go away, but it was in the path of least mental overhead to get from the massive function in manifest to these smaller structs.